### PR TITLE
#245 mark as read a message only if is unread

### PR DESCRIPTION
**Pull request by oubchid**

### CODE REVIEW

**Review by rouazana — COMMENTED** [review]
why didn't you add a test to match the new case?

**Comment by MichaelBailly** [comment]
a test please

**Comment by oubchid** [comment]
I'm adding a condition here 
so What is the test to add in this case ?

**Comment by fabienmoyon** [comment]
> I'm adding a condition here
> so What is the test to add in this case ?

Like @rouazana said, add one more that verify your condition : `should not call markAsRead if the email is already read`

**Review by rezk2ll — APPROVED** [review]

**Comment by oubchid** [comment]
@rouazana @michaelbailly tests added

**Review by fabienmoyon — APPROVED** [review]

File: src/linagora.esn.unifiedinbox/app/controllers.spec.js [hunk 1]
```diff
@@ -448,7 +449,7 @@ describe('The linagora.esn.unifiedinbox module controllers', function() {
       });
       jmapClient.getMessages = function() {
         return $q.when([{
-          isFlagged: true, textBody: 'textBody', htmlBody: 'htmlBody', attachments: []
+          isUnread: false, isFlagged: true, textBody: 'textBody', htmlBody: 'htmlBody', attachments: []
```
Comment (rouazana) [inline src/linagora.esn.unifiedinbox/app/controllers.spec.js:452]: why do you need this then?

Comment (oubchid) [reply src/linagora.esn.unifiedinbox/app/controllers.spec.js:452]: cuz now with new case, test failed and i fix it it by keeping `isUnread` to false (opening item from the list => isUnread should be false) to match this one 
```
expect(after).to.shallowDeepEqual({
      `isUnread: false`, id: 'messageId1', isFlagged: true, textBody: 'textBody', htmlBody: 'htmlBody', attachments: [], loaded: true
 });
```

**Pull request merged**